### PR TITLE
Improved next/previous tip logic, closes #7

### LINKED ISSUE #7: Fix previous tip
The current implementation is quite poor and a better solution will need to be found.

Ideas are welcome.

### PR DESCRIPTION
**Pull request by AkshayAgarwal007**

### CODE REVIEW

**Comment by AkshayAgarwal007** [comment]
Incorporating the idea by @humdingerb  in #7 .

File: source/Shuffle.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+
+void 
+Randomize(int* arr, int n)
+{
+	srand ( time(NULL) );
+
+	for(int i = n - 1; i > 0; i--) {
+		int j = rand() % (i+1);
```
Comment (scottmc) [inline source/Shuffle.cpp:23]: gcc2 required variables to be declared prior to code (c89).  I suspect this code will fail if you were to try and build it with gcc2.  Looking at the haikuports recipe, the 1.0.1 version was marked as compiling on both gcc2 and gcc4.

Comment (pulkomandy) [reply source/Shuffle.cpp:23]: Only in C code. C++ allowed this from the start and gcc2 handles it fine there.

File: source/Shuffle.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ */
+#ifndef SHUFFLE_H
+#define SHUFFLE_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+
+void Swap(int* a, int* b);
```
Comment (pulkomandy) [inline source/Shuffle.h:13]: Any reason for doing this with your own code rather than using http://en.cppreference.com/w/cpp/algorithm/random_shuffle ?

Comment (AkshayAgarwal007) [reply source/Shuffle.h:13]: random_shuffle has got deprecated in the latest revisions and shuffle won't work with gcc2.
This one makes use of random_shuffle, just tried in my test branch...
https://github.com/AkshayAgarwal007/Tipster/commit/c7e91640962b01466aa59f92a1c140a9f834bfcd

**Pull request merged**